islandora_transforms/BPL_MODS_to_solr.xslt

```diff
@@ -176,6 +177,43 @@
     <xsl:apply-templates mode="slurp_for_bpl"/>
   </xsl:template>
 
+  <!-- Fields created from the non-marc-call-number. -->
+  <xsl:template mode="slurp_for_bpl" match="mods:mods/mods:identifier[@type = 'non-marc-call-number']">
+    <xsl:variable name="content" select="normalize-space(text())"/>
+    <xsl:if test="not($content = '')">
+      <!-- XXX: In an ideal world, we'd be on Solr 6+ and able to use query slop
+         to force token ordering at query time with fancy padding. Instead,
+         manually tokenize, set sort padding, and re-assemble/store so we can
+         use an untokenized string to sort at query time. -->
+      <xsl:call-template name="write_bpl_field">
+        <xsl:with-param name="field_name" select="'identifier_non_marc_call_number'"/>
+        <xsl:with-param name="content">
+          <xsl:for-each select="exslt-string:tokenize($content, '-_ ')">
+            <xsl:variable name="token" select="normalize-space(.)"/>
+            <xsl:choose>
+              <xsl:when test="number($token) = number($token)">
+                <xsl:value-of select="format-number($token, 000000)"/>
```

Comment:
Doesn't appear to do the trick:
```json
{
  "responseHeader":{
    "status":0,
    "QTime":0,
    "params":{
      "q":"PID:\"islandora:3\"",
      "indent":"true",
      "fl":"PID,bpl_*",
      "wt":"json"}},
  "response":{"numFound":1,"start":0,"docs":[
      {
        "PID":"islandora:3",
        "bpl_mods_identifier_non-marc-filing-suffix_concatenated_non-marc-call-number_ms":["C-123-345"],
        "bpl_mods_identifier_non-marc-filing-suffix_concatenated_non-marc-call-number_mt":["C-123-345"],
        "bpl_mods_titleInfo_title_concatenated_nonSort_ms":["asdf"],
        "bpl_mods_titleInfo_title_concatenated_nonSort_mt":["asdf"],
        "bpl_mods_titleInfo_title_concatenated_nonSort_concatenated_subTitle_ms":["asdf"],
        "bpl_mods_titleInfo_title_concatenated_nonSort_concatenated_subTitle_mt":["asdf"],
        "bpl_mods_all_call_numbers_ms":["L-3422-234",
          "C-123-345"],
        "bpl_mods_all_call_numbers_mt":["L-3422-234",
          "C-123-345"],
        "bpl_mods_all_call_numbers_est":["L-3422-234",
          "C-123-345"],
        "bpl_mods_identifier_non_marc_call_number_ss":"C123345"}]
  }}
```

Reply:
```bpl_mods_identifier_non_marc_call_number_ss``` should be ```C000123000345```, if we were to pad each numerical component to 6 digits...

Reply:
Just double checking, making sure it is actually running through the number format bit:
```
[...]
              <xsl:when test="number($token) = number($token)">
                <xsl:text>blep</xsl:text>
                <xsl:value-of select="format-number($token, 000000)"/>
              </xsl:when>
[...]
```
giving:
```
        "bpl_mods_identifier_non_marc_call_number_ss":"Cblep123blep345"}]
```